manifests/init.pp

```diff
@@ -1,21 +1,35 @@
 class wireguard (
-  Array[Hash] $tunnels = [],
-  Array[Hash] $simple_tunnels = [],
+  Hash[String, Struct[
```

Comment:
Is it possible to just add below parameter validation only inside `wireguard::tunnel`? This would avoid duplication.